test/saveHashTable.m

```diff
@@ -1,37 +1,163 @@
-function saveHashTable(status)
+function saveHashTable(status, varargin)
 % SAVEHASHTABLE saves the references hashes for the Matlab2Tikz tests
 %
 % Usage:
 %  SAVEHASHTABLE(status)
 %
+%  SAVEHASHTABLE(status, 'dryrun', BOOL, ...) determines whether or not to
+%  write the constructed hash table to file (false) or to stdout (true).
+%  Default: false
+%
+%  SAVEHASHTABLE(status, 'removedTests', CHAR, ...) specifies which action to
+%  execute on "removed tests" (i.e. test that have a hash recorded in the file,
+%  but which are not present in `status`). Three values are possible:
+%   - 'ask' (default): Ask what to do for each such test.
+%   - 'remove': Remove the test from the file.
+%               This is appropriate if the test has been removed from the suite.
+%   - 'keep': Keep the test hash in the file.
+%             This is appropriate when the test has not executed all tests.
+%
 % Inputs:
 %   - status: output cell array of the testing functions
 %
 % See also: runMatlab2TikzTests, testMatlab2tikz
+    ipp = m2tInputParser();
+    ipp = ipp.addRequired(ipp, 'status', @iscell);
+    ipp = ipp.addParamValue(ipp, 'dryrun', false, @islogical);
+    ipp = ipp.addParamValue(ipp, 'removedTests', 'ask', @isValidAction);
+    ipp = ipp.parse(ipp, status, varargin{:});
+
+    %% settings
     suite = status{1}.testsuite; %TODO: handle multiple test suites in a single array
     filename = hashTableName(suite);
+    FILEFORMAT = '%s : %s\n';
```

Comment:
In Win this is a problem, it should be `\r\n`. However, you can actually skip it altogether and let textscan deal with it internally, i.e. `'%s : %s'` should work OS independently.